src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,33 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+      Animal.alive = Animal.alive.filter((item) => item.health > 0);
+    }
```

Comment:
The `bite` method currently only filters the `Animal.alive` array when a Carnivore successfully bites a Herbivore. However, the task requires that an animal should be removed from the `Animal.alive` array when its health becomes less than or equal to 0. This logic should not be tied only to the biting action. Consider moving the logic for removing dead animals from the `Animal.alive` array to a separate method within the Animal class, which can be called whenever an animal's health changes.